projects/sdk-wells/src/lib/Well.ts

```
@@ -45,28 +56,46 @@ export class Well {
   }
 
   /**
-   * Retrieves all the well data and stores it in the Well object. Future
-   * getXXX() calls will retrive the local data instead of making an RPC call.
-   * Effectively caches data for:
-   * - getName()
-   * - getWell()
-   * - getLPToken()
-   * - getTokens()
-   * - getWellFunction()
-   * - getPumps()
-   * - getAuger()
-   * - getReserves()
-   * */
-  async loadWell() {
-    await Promise.all([this.getName(), this.getLPToken(), this.getWell(), this.getReserves()]);
+   * Loads Well data from chain
+   *
+   * If no options are specified, it will load everything. However, if
+   * an options object is passed, it will only load those the data
+   * whose options is set to true.
+   *
+   * loadWell() -- loads everything
+   * loadWell({tokens: true}) - only loads tokens
+   *
+   */
+  async loadWell(options?: LoadOptions): Promise<void> {
+    // TODO: use a multicall
+    const toLoad = [];
+
+    if (!options) {
+      toLoad.push(this.getName(), this.getLPToken(), this.getWell());
+    } else {
+      // console.log("load", options);
+      if (options.name) toLoad.push(this.getName());
+      if (options.lpToken) toLoad.push(this.getLPToken());
+      if (options.tokens || options.wellFunction || options.pumps || options.auger) toLoad.push(this.getWell());
+    }
+
+    await Promise.allSettled(toLoad);
```

Do we want it to fail silently? I think we'd prefer `Promise.all` instead.